### PR TITLE
Adding v5 docblock

### PR DESCRIPTION
**Pull request by nyamsprod**
- adding tests to cover all new Object and methods
- improve docblocks
- normalize class file headers
- improve classes typehint and return type

### CODE REVIEW

**Comment by nyamsprod** [comment]
@jeremykendall I tried to not add the "controversial point of the previous PR" I've only updated docblock and testsuite. Hope you'll like it better 👍

File: src/Pdp/Exception/SeriouslyMalformedUrlException.php [hunk 1]
```diff
@@ -1,13 +1,14 @@
 <?php
-
 /**
- * PHP Domain Parser: Public Suffix List based URL parsing.
+ * Public Suffix List PHP: Public Suffix List based URL parsing.
```
Comment (jeremykendall) [inline src/Pdp/Exception/SeriouslyMalformedUrlException.php]: These should all change back to "PHP Domain Parser: Public Suffix List based URL parsing."

File: src/Pdp/LabelsTrait.php [hunk 2]
```diff
@@ -1,34 +1,61 @@
 <?php
-
-declare(strict_types=1);
```
Comment (jeremykendall) [inline src/Pdp/LabelsTrait.php]: Shouldn't all these files include this `declare` statement?

**Comment by jeremykendall** [comment]
Added a few comments but I'm withholding a "real" review until I'm not so tired. I think I got 3 hours sleep last night because kiddo didn't sleep.

File: src/Pdp/Exception/SeriouslyMalformedUrlException.php (hunk 1 above)
Comment (jeremykendall) [reply src/Pdp/Exception/SeriouslyMalformedUrlException.php]: (I used "Public Suffix List PHP" for the new library.)

Comment (nyamsprod) [reply src/Pdp/Exception/SeriouslyMalformedUrlException.php]: yep .. I'll updatethat

File: src/Pdp/LabelsTrait.php (hunk 2 above)
Comment (nyamsprod) [reply src/Pdp/LabelsTrait.php]: yep typo from my part it will be change too

Comment (nyamsprod) [reply src/Pdp/LabelsTrait.php]: declare is present just moved after the credits

**Comment by nyamsprod** [comment]
a few questions/comments:

- the `PublicSuffixList` constructor does too many things IMHO this is a perfect use case for some named constructors. Something like `PublicSuffixList::createFromPath` /`PublicSuffixList::fromPath` to simplify the constructor signature `PublicSuffixList::__construct(array $rules = null)` .

- `MatchDomain` and `UnmatchedDomain` don't really need the constructor `$isValid` argument since the domain status is already determined by the class instance.

File: tests/src/Pdp/HttpAdapter/CurlHttpAdapterTest.php [hunk 3]
```diff
@@ -30,7 +30,7 @@ protected function tearDown()
 
     public function testGetContent()
     {
-        $content = $this->adapter->getContent('http://www.google.com');
+        $content = $this->adapter->getContent('https://www.google.com');
```
Comment (jeremykendall) [inline tests/src/Pdp/HttpAdapter/CurlHttpAdapterTest.php]: 👍

**Pull request merged**